sdk/src/main/java/com/bugsnag/android/ForegroundDetector.java

```diff
@@ -0,0 +1,62 @@
+package com.bugsnag.android;
+
+import android.app.ActivityManager;
+import android.content.Context;
+import android.os.Build;
+import android.os.Process;
+import android.support.annotation.Nullable;
+
+import java.util.List;
+
+class ForegroundDetector {
+
+    private final ActivityManager activityManager;
+
+    ForegroundDetector(Context context) {
+        this.activityManager =
+            (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+    }
+
+    boolean isInForeground() {
+        ActivityManager.RunningAppProcessInfo info = getProcessInfo();
+
+        if (info != null) {
+            return info.importance <= ActivityManager.RunningAppProcessInfo.IMPORTANCE_VISIBLE;
+        } else { // prefer a potential false positive if process info not available
+            return true;
```

Comment:
Why default rather than falling back to the old implementation?

Reply:
We made the decision when implementing ANR detection that we should prefer a false positive when we're unable to determine process importance. This is because otherwise the notifier might discard an actual error, since the old implementation can be inaccurate if the main thread is blocked and lifecycle callbacks cannot be invoked.

Reply:
I see. To rephrase, there are actually three states for whether the app is in the foreground, `yes`, `no`, and `unknown`. In the `unknown` case, ANR errors should still be reported, so this uses `yes` instead of `unknown` or an implementation based on main thread activity. Is that a fair assessment? 

It would be clearer (and delineate responsibilities better) to actually return a value which indicates that the state is currently unknown and let the ANR reporter handle that appropriately, but barring that, the comment should state for future travelers why a potential false positive is preferable and how it affects reporting. "If this was false, some ANRs might get ignored" was not what I expected, and I can see how a future PR to tweak to how this value is computed could unintentionally break this.